src/library/Uno.Material/Extensions/ControlExtensions.cs

```diff
@@ -4,37 +4,116 @@
 using System.Diagnostics.CodeAnalysis;
 using Windows.UI;
 using Uno.Disposables;
+using System.Windows.Input;
 
 #if WinUI
 using Microsoft.UI.Xaml;
 using Microsoft.UI.Xaml.Media;
 using Microsoft.UI.Xaml.Data;
 using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
 #else
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Media;
 using Windows.UI.Xaml.Data;
 using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
 #endif
 
 namespace Uno.Material
 {
 	[Bindable]
 	public static class ControlExtensions
 	{
-		#region DependencyProperty: Icon
+		#region DependencyProperty: LeadingIcon
 
-		public static DependencyProperty IconProperty { [DynamicDependency(nameof(GetIcon))] get; } = DependencyProperty.RegisterAttached(
-			"Icon",
+		public static DependencyProperty LeadingIconProperty { [DynamicDependency(nameof(GetLeadingIcon))] get; } = DependencyProperty.RegisterAttached(
+			"LeadingIcon",
 			typeof(IconElement),
 			typeof(ControlExtensions),
 			new PropertyMetadata(default));
 
-		[DynamicDependency(nameof(SetIcon))]
-		public static IconElement GetIcon(Control obj) => (IconElement)obj.GetValue(IconProperty);
+		[DynamicDependency(nameof(SetLeadingIcon))]
+		public static IconElement GetLeadingIcon(Control obj) => (IconElement)obj.GetValue(LeadingIconProperty);
 
-		[DynamicDependency(nameof(GetIcon))]
-		public static void SetIcon(Control obj, IconElement value) => obj.SetValue(IconProperty, value);
+		[DynamicDependency(nameof(GetLeadingIcon))]
+		public static void SetLeadingIcon(Control obj, IconElement value) => obj.SetValue(LeadingIconProperty, value);
+
+		#endregion
+
+		#region DependencyProperty: IsLeadingIconVisible
+
+		public static DependencyProperty IsLeadingIconVisibleProperty { [DynamicDependency(nameof(GetIsLeadingIconVisible))] get; } = DependencyProperty.RegisterAttached(
+			"IsLeadingIconVisible",
+			typeof(bool),
+			typeof(ControlExtensions),
+			new PropertyMetadata(true));
+
+		[DynamicDependency(nameof(SetIsLeadingIconVisible))]
+		public static bool GetIsLeadingIconVisible(Control obj) => (bool)obj.GetValue(IsLeadingIconVisibleProperty);
+
+		[DynamicDependency(nameof(GetIsLeadingIconVisible))]
+		public static void SetIsLeadingIconVisible(Control obj, bool value) => obj.SetValue(IsLeadingIconVisibleProperty, value);
+
+		#endregion
+
+		#region DependencyProperty: LeadingIconCommand
```

Comment:
What do we think of changing the names for `LeadingIconCommand` and `TrailingIconCommand` to just `LeadingCommand` and `TrailingCommand`? @agneszitte / @Xiaoy312 ?

Reply:
LeadingCommand/TrailingCommand work for me.
@kazo0  Should we have TextBoxExtensions.LeadingCommand/TrailingCommand to match your other comment then?

Reply:
yep

Reply:
Done